src/sender.cpp

```diff
@@ -54,28 +56,49 @@ Sender::Sender(const std::string &interface_name, uint16_t caracal_id)
   }
 
   // Find the IPv4/v6 gateway.
-  Tins::NetworkInterface interface { interface_name };
+  Tins::NetworkInterface interface { config.interface };
   Tins::HWAddress<6> gateway_mac{"00:00:00:00:00:00"};
   if (l2_protocol_ == Protocols::L2::Ethernet) {
     spdlog::info("Resolving the gateway MAC address...");
-    gateway_mac =
-        Utilities::gateway_mac_for(interface, Tins::IPv4Address("8.8.8.8"));
+    if (config.ip_version == 6) {
```

Comment:
In the future it might be best to resolve both gateway automatically and use them depending on the probe protocol. E.g. set `gateway_mac_v4` and `gateway_mac_v6`. In this case the `-v4/v6` flags/options are probably not needed anymore?

Can you add this comment in the code here so that we remember it?